sedbackend/apps/cvs/simulation/storage.py

```diff
@@ -463,6 +471,27 @@ def parse_formula(formula: str, vd_values, mi_values):
                     'mi' + id, mi['value'], new_formula)
 
     return new_formula
+'''
+
+
+def parse_formula(formula: str, vd_values, mi_values) -> str:
+    new_formula = formula
+    vd_names = expr.get_prefix_variables('VD', new_formula)
+    mi_names = expr.get_prefix_variables('EF', new_formula)
+
+    for vd in vd_values:
+        for name in vd_names:
+            unit = vd["unit"] if vd["unit"] is not None and vd["unit"] != "" else "N/A"
+            if name == f'{vd["name"]} [{unit}]':
+                new_formula = expr.replace_prefix_variables("VD", name, str(vd["value"]), new_formula)
+    for mi in mi_values:
+        for name in mi_names:
+            unit = mi["unit"] if mi["unit"] is not None and mi["unit"] != "" else "N/A"
+            if name == f'{mi["name"]} [{unit}]':
+                new_formula = expr.replace_prefix_variables("EF", name, str(mi["value"]), new_formula)
+    new_formula = expr.remove_strings_replace_zero(new_formula)
```

Comment:
I don't know if this line but something in the parsing does not work right.

Got the following error with the given input:
![Screenshot_2023-04-05_22-21-57](https://user-images.githubusercontent.com/57296415/230183714-bcd32a5a-afe7-4f1e-be25-c0779e873da7.png)
![Screenshot_2023-04-05_22-22-10](https://user-images.githubusercontent.com/57296415/230183722-5fb25e60-ac89-4b63-ba19-dc25648ca143.png)